libs/langchain/langchain/storage/file_system.py

```diff
@@ -0,0 +1,118 @@
+import re
+from pathlib import Path
+from typing import Iterator, List, Optional, Sequence, Tuple, Union
+
+from langchain.schema import BaseStore
+from langchain.storage.exceptions import InvalidKeyException
+
+
+class LocalFileStore(BaseStore[str, bytes]):
+    """AbstractStore interface that works on the local file system.
+
+    Examples:
+        Create a LocalFileStore instance and perform operations on it:
+
+        .. code-block:: python
+
+            # Instantiate the LocalFileStore with the root path
+            file_store = LocalFileStore("/path/to/root")
+
+            # Set values for keys
+            file_store.mset([("key1", b"value1"), ("key2", b"value2")])
+
+            # Get values for keys
+            values = file_store.mget(["key1", "key2"])  # Returns [b"value1", b"value2"]
+
+            # Delete keys
+            file_store.mdelete(["key1"])
+
+            # Iterate over keys
+            for key in file_store.yield_keys():
+                print(key)
+
+    """
+
+    def __init__(self, root_path: Union[str, Path]) -> None:
+        """Implement the AbstractStore interface for the local file system.
+
+        Args:
+            root_path (Union[str, Path]): The root path of the file store. All keys are
+                                          interpreted as paths relative to this root.
```

Comment:
not to über nitpick, but this is how google style guide indents 🌝 
![Screenshot 2023-08-08 at 1 10 28 PM](https://github.com/langchain-ai/langchain/assets/22008038/4ece2c75-d9f5-4af0-ab59-8def28d0eda2)

Reply:
that works for me